packages/frontend/src/model/eventlogs.ts

```diff
@@ -1,49 +1,87 @@
+import { ApiAuthGet, useAuthApiQuery } from '@/model/api';
 import { makeApiAuthClient } from '@/utils/api';
-import { EventLogDto } from 'api/dist/eventlog/types';
+import { EventLogDto, EventLogTypeDto } from 'api/dist/eventlog/types';
 import { PaginatedResponseBody } from 'api/dist/shared/types';
+import { AxiosResponse } from 'axios';
 import { useEffect, useState } from 'react';
+import { atom, selector } from 'recoil';
 
 export type AllEventLogsQueryParameters = {
   sortColumn?: string;
   sortType?: string;
   limit?: number;
   page?: number;
+  type?: string;
+  search?: string;
 };
 
-export const useAllEventLogs = ({
-  page,
-  limit,
-  sortColumn,
-  sortType,
-}: AllEventLogsQueryParameters): {
+export const eventLogsQueryParameters = atom({
+  key: 'QueryParameters',
+  default: {
+    sortColumn: 'createdAt',
+    sortType: 'desc',
+    limit: 15,
+    page: 1,
+    type: '',
+    search: '',
+  },
+});
+
+/**
+ * Query selector that fetches all praise periods from the API.
+ */
+export const AllEventLogsQuery = selector({
+  key: 'AllEventLogsQuery',
+  get: ({ get }): AxiosResponse<unknown> => {
+    const query = get(eventLogsQueryParameters);
+
+    const qs = Object.keys(query)
+      .map((key) => `${key}=${query[key]}`)
+      .join('&');
+
+    const response = get(
+      ApiAuthGet({
+        url: `/eventlogs/all${qs ? `?${qs}` : ''}`,
```

Comment:
It might be a bit cleaner to use the `apiAuthClient` instead of the  `ApiAuthGet` recoil selector --  then you can pass the query parameters as an object and you don't have to do this string formatting stuff.